plugins/org.polarsys.capella.cybersecurity.sirius.analysis/src/org/polarsys/capella/cybersecurity/sirius/analysis/CybersecurityServices.java

```diff
@@ -616,12 +792,70 @@ public String getAssetDecorationBorderSize(DSemanticDecorator view) {
             return "5"; //$NON-NLS-1$
           }
         }
+        for (Threat threat : getRelatedThreats(e)) {
+          DDiagramElement element = DiagramServices.getDiagramServices().getDiagramElement(diagram, threat);
+          if (element instanceof DNode) {
+            return "5"; //$NON-NLS-1$
+          }
+        }
+      }
+      return "1"; //$NON-NLS-1$
+    }
+    return null;
+  }
+
+  public String getSupportingAssetsDecorationBorderSize(DSemanticDecorator view) {
+    if (view instanceof AbstractDNode) {
+      EObject e = view.getTarget();
+
+      if (!mustHighlightComponent(e)) {
+        return "1";
+      }
+
+      DDiagram diagram = CapellaServices.getService().getDiagramContainer(view);
+      if (diagram != null) {
+        for (PrimaryAsset asset : getRelatedAssets(e)) {
+          DDiagramElement element = DiagramServices.getDiagramServices().getDiagramElement(diagram, asset);
+          if (element instanceof DNode) {
+            return "5"; //$NON-NLS-1$
+          }
+        }
+        for (Threat threat : getRelatedThreats(e)) {
+          DDiagramElement element = DiagramServices.getDiagramServices().getDiagramElement(diagram, threat);
+          if (element instanceof DNode) {
+            return "5"; //$NON-NLS-1$
+          }
+        }
       }
       return "1"; //$NON-NLS-1$
     }
     return null;
   }
 
+  private boolean mustHighlightComponent(EObject element) {
+    if (!(element instanceof Part)) {
+      return false;
+    }
+    if (((Part) element).getAbstractType() instanceof PhysicalComponent) {
+      PhysicalComponent comp = (PhysicalComponent) ((Part) element).getAbstractType();
+      if (comp.getNature() == PhysicalComponentNature.NODE) {
+        return true;
+      }
+    }
+
+    Component component = (Component) ((Part) element).getAbstractType();
+    List<AbstractFunction> allocatedFunctions = component.getAllocatedFunctions();
+    return allocatedFunctions.size() > 0 ? true : false;
+  }
+
+  private boolean mustHighlightExchange(EObject element) {

```

Comment:
can you add "SupportingAsset" word in this method, to tell that this method is used for SupportingAsset layer, 
`isSupportingAssetHighlightedExchange` something like that

Reply:
otherwise looks perfect !